### PR TITLE
Metric name doc

### PR DESCRIPTION
**Pull request by bdunavant**

### CODE REVIEW

File: content/irondb/metric-names-tags-queries.md [hunk 1]
```diff
@@ -1,9 +1,48 @@
...
+Examples:
+ * my_metric_name
+ * my_metric_name|ST[color:blue,env:prod]
+ * my_metric_name|MT{}|ST[env:prod]|MT{foo}|ST[color:blue]  
+
+The final example would canonicalize into the previous example since measurement-tags are not currently stored.
+
+### Metric Names
+
+Metric names in Circonus may be an string of bytes other than a null character, or the stream-tag or measurement-tags identifiers ("|ST[" or "|MT{").
```
[the remark this answers is not in the document]
Comment (bdunavant) [reply content/irondb/metric-names-tags-queries.md:35]: My understanding is literally any byte other than \0

File: content/irondb/metric-names-tags-queries.md [hunk 2]
```diff
@@ -1,9 +1,48 @@
...
+<metric-name> ::= <characters>
+<tag-section> ::= (<stream-tags> | <measurement-tags>)*
+<stream-tags> ::= "|ST[" <tagset> "]" | ""
+<measurement-tags> ::= "|MT{" <tagset> "}" | ""
+<tagset> ::= <tag> "," <tagset> | <tag> | ""
+<tag>  ::= <tag-category> ":" <tag-value> | <tag-category>
+
+To be canonical:
+ * A full canonical metric name must be less than 4095 characters in length.
+ * <tagsets> must have duplicate <tag> items removed, and then sorted lexically by category, and then value.
```
[the remark this answers is not in the document]
Comment (bdunavant) [reply content/irondb/metric-names-tags-queries.md]: I have put back-ticks.

File: content/irondb/metric-names-tags-queries.md [hunk 3]
```diff
@@ -1,9 +1,48 @@
...
+To be canonical:
+ * A full canonical metric name must be less than 4095 characters in length.
+ * <tagsets> must have duplicate <tag> items removed, and then sorted lexically by category, and then value.  
+
+Submissions will be canonicalized before storage.
+
+Examples:
+ * my_metric_name
+ * my_metric_name|ST[color:blue,env:prod]
+ * my_metric_name|MT{}|ST[env:prod]|MT{foo}|ST[color:blue]
```
[the remark this answers is not in the document]
Comment (bdunavant) [reply content/irondb/metric-names-tags-queries.md]: I have put back-ticks.

File: content/irondb/metric-names-tags-queries.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (esproul) [reply content/irondb/metric-names-tags-queries.md]: Also, would use "lexicographically" as it's the correct term here.

Comment (esproul) [inline content/irondb/metric-names-tags-queries.md]:
```suggestion
 * <tagsets> must have duplicate <tag> items removed, and then sorted lexicographically by category, and then value.  
```

File: content/irondb/metric-names-tags-queries.md [hunk 4]
```diff
@@ -1,9 +1,48 @@
...
+Examples:
+ * `my_metric_name`
+ * `my_metric_name|ST[color:blue,env:prod]`
+ * `my_metric_name|MT{}|ST[env:prod]|MT{foo}|ST[color:blue]`
+
+The final example would canonicalize into the previous example since measurement-tags are not currently stored.
+
+### Metric Names
+
+Metric names in Circonus may be an string of bytes other than a null character, or the stream-tag or measurement-tags identifiers ("|ST[" or "|MT{").
```
Comment (esproul) [inline content/irondb/metric-names-tags-queries.md:35]:
```suggestion
Metric names in Circonus may be an string of bytes other than a null character, or the stream-tag or measurement-tags identifiers (`|ST[` or `|MT{`).
```

Easier to parse visually when rendered in a non-proportional font.

**Review by esproul — APPROVED** [review]
LGTM

**Pull request merged**